src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -143,31 +154,27 @@
    pool :- jruby-schemas/pool-queue-type]
   (.pollFirst pool timeout TimeUnit/MILLISECONDS))
 
+(schema/defn borrowed-poison-pill
+  [pool :- jruby-schemas/pool-queue-type
+   instance :- PoisonPill]
+  (.putFirst pool instance)
+  (throw (IllegalStateException.
```

Comment:
I suggest deleting this function and moving its code back into the `cond` below - that's the only place its used anyway, and as @nwolfe, it sure looks baffling out of context.

Unless this needs to be reused?